src/main/java/aquality/selenium/core/elements/Element.java

```diff
@@ -0,0 +1,145 @@
+package aquality.selenium.core.elements;
+
+import aquality.selenium.core.applications.IApplication;
+import aquality.selenium.core.configurations.IElementCacheConfiguration;
+import aquality.selenium.core.elements.interfaces.*;
+import aquality.selenium.core.localization.ILocalizedLogger;
+import aquality.selenium.core.logging.Logger;
+import aquality.selenium.core.utilities.IElementActionRetrier;
+import aquality.selenium.core.waitings.IConditionalWait;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.remote.RemoteWebElement;
+
+import java.util.function.Supplier;
+
+public abstract class Element implements IElement {
+
+    private final String name;
+    private final ElementState elementState;
+    private final By locator;
+    private IElementCacheHandler elementCacheHandler;
+
+    protected Element(final By loc, final String name, final ElementState state) {
+        locator = loc;
+        this.name = name;
+        elementState = state;
+    }
+
+    protected abstract IApplication getApplication();
+
+    protected abstract IElementFactory getElementFactory();
+
+    protected abstract IElementFinder getElementFinder();
+
+    protected abstract IElementCacheConfiguration getElementCacheConfiguration();
+
+    protected abstract IElementActionRetrier getElementActionRetrier();
+
+    protected abstract ILocalizedLogger getLocalizedLogger();
+
+    protected abstract IConditionalWait getConditionalWait();
+
+    protected abstract String getElementType();
+
+    protected IElementCacheHandler getCache() {
+        if (elementCacheHandler == null) {
+            elementCacheHandler = new ElementCacheHandler(locator, elementState, getElementFinder());
+        }
+
+        return elementCacheHandler;
+    }
+
+    protected Logger getLogger() {
+        return Logger.getInstance();
+    }
+
+    protected ElementState getElementState() {
```

Comment:
do we need it to be public? we don't have it in dotnet

Reply:
it was in selenium-java
I am not sure, lets remove it